syntax = "proto3";

package json_person;

message Person {
  string name = 1;
  int32 id = 2;
  string email = 3;

  enum PhoneType { MOBILE = 0; HOME = 1; WORK = 2; }

  message PhoneNumber {
    string number = 1 [json_name = "phone_number"];
    PhoneType type = 2 [json_name = "phone_type"];
  }

  PhoneNumber phone = 4;
}